tests/test_modules/test_web/test_system_websocket.py

```diff
@@ -26,31 +26,136 @@ def tearDown(self):
-        req = dict(
-            typeid="malcolm:core/Post:1.0",
-            id=0,
-            path=["hello", "greet"],
-            parameters=dict(
-                name="me"
-            )
-        )
         conn.write_message(json.dumps(req))
         resp = yield conn.read_message()
         resp = json.loads(resp)
         self.result.put(resp)
         conn.close()
 
+    @gen.coroutine
+    def send_non_JSON_message(self):
+        conn = yield websocket_connect("ws://localhost:%s/ws" % self.socket)
+        conn.write_message("I am not JSON!")
+        resp = yield conn.read_message()
+        resp = json.loads(resp)
+        self.result.put(resp)
+        conn.close()
+
     def test_server_and_simple_client(self):
-        self.server._loop.add_callback(self.send_message)
+        self.server._loop.add_callback(self.send_message,
+                                       dict(
+                                            typeid="malcolm:core/Post:1.0",
+                                            id=0,
+                                            path=["hello", "greet"],
+                                            parameters=dict(
+                                                name="me"
+                                            )
+                                       )
+        )
+
         resp = self.result.get(timeout=2)
         assert resp == dict(
             typeid="malcolm:core/Return:1.0",
             id=0,
             value="Hello me"
         )
 
+    def test_error_server_and_simple_client_badJSON(self):
+        self.server._loop.add_callback(self.send_message, "I am JSON (but not a dict)")
+        resp = self.result.get(timeout=2)
+        assert resp == dict(
+            typeid="malcolm:core/Error:1.0",
+            id=-1,
+            message="ValueError: Error decoding JSON object (didn't return OrderedDict)"
+        )
+
+        self.server._loop.add_callback(self.send_non_JSON_message)
+        resp = self.result.get(timeout=2)
+        assert resp == dict(
+            typeid="malcolm:core/Error:1.0",
+            id=-1,
+            message="ValueError: Error decoding JSON object (No JSON object could be decoded)"
+        )
+
+    def test_error_server_and_simple_client(self):
+        self.server._loop.add_callback(self.send_message,
+                                       dict(
+                                           typeid="malcolm:core/Post:1.0",
+                                           path=["hello", "greet"],
+                                           parameters=dict(
+                                               name="me"
+                                                )
+                                           )
+                                       )
+        resp = self.result.get(timeout=2)
+        assert resp == dict(
+            typeid="malcolm:core/Error:1.0",
+            id=-1,
+            message="KeyError: 'id field not present in JSON message'"
```

Comment:
To get rid of these extra quotes we need to change serialize_hook in serializable.py to return:

`"%s: %s" % (type(o).__name__, o.message)`

instead of 

`"%s: %s" % (type(o).__name__, o)`